src/main/java/com/bathanh/exercise/oop/oop1/Triangle.java

```diff
@@ -0,0 +1,55 @@
+package com.bathanh.exercise.oop.oop1;
+
+public class Triangle implements Shape {
+
+    private Point point1;
+    private Point point2;
+    private Point point3;
+
+    public Triangle(Point point1, Point point2, Point point3) {
+        this.point1 = point1;
+        this.point2 = point2;
+        this.point3 = point3;
+    }
+
+    public double getArea() {
+        double x1 = point1.getX();
+        double y1 = point1.getY();
+        double x2 = point2.getX();
+        double y2 = point2.getY();
+        double x3 = point3.getX();
+        double y3 = point3.getY();
+
+        return Math.abs(0.5 * ((x2 - x1) * (y3 - y1) - (x3 - x1) * (y2 - y1)));
+    }
+
+    public double calculateDistance(Point point1, Point point2) {
```

Comment:
Hàm này có thể được khai báo ngay trong class Point
```java
public double distanceTo(final Point point) {
...
}
```
và dùng

Reply:
tham số đầu vào phải là 2 điểm để mình tính khoảng cách chứ a

Reply:
@hobathanh nếu đặt nó trong class Point thì this là 1 point rồi

Reply:
resolved